### ISSUE TITLE
Sorting with .populate() option

### ISSUE THREAD

**Issue by Joanne16** [balderdashy/waterline#1245]
Hi, is it possible to apply sorting with associations ? For example if we have the following:

model employeeModel
{...
    attributes: {
id: {
            type: 'integer',
            primaryKey: true,
            autoIncrement: true,
            required: true
        },
        companyId: {
            model: 'companyModel'
            type: 'integer',
            required: true,
            defaultsTo: 0
        }
}
}

What I need as results is a list of employees sorted by the company name, where the company's name is in the 'companyModel'. The following does not work:

model.find({
         where: {},
         sort: 'companyId.companyName'
    })
    .populate('companyId')
    .then(function (result) {
         callback(null, {rows: result});
      })
   .catch(function (err) {
         // an error occurred
         return callback(err);
     });

Is it possible to sort the query results by a populated field ?

Thanks.

**Comment by devinivy** [comment]
Currently there is no way to do this.  @atiertant, will deep-populate help with this?

**Comment by atiertant** [comment]
@devinivy no this would be is a deepCriteria
this is that feature request https://github.com/balderdashy/waterline/issues/266
but a deep-populate would be the first step ...

**Comment by sailsbot** [comment]
Thanks for posting, @Joanne16.  I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!

**Comment by Tronix117** [comment]
Should not be closed
